components/raftstore/src/store/snap/io.rs

```diff
@@ -36,15 +43,45 @@ pub fn build_plain_cf_file<E>(
 where
     E: KvEngine,
 {
-    let mut file = box_try!(OpenOptions::new().write(true).create_new(true).open(path));
+    let mut file = Some(box_try!(OpenOptions::new()
+        .write(true)
+        .create_new(true)
+        .open(path)));
+    let mut encrypted_file: Option<EncrypterWriter<File>> = None;
+    if let Some(key_mgr) = key_mgr {
+        let enc_info = box_try!(key_mgr.new_file(path));
+        let mthd = encryption_method_from_db_encryption_method(enc_info.method);
+        if mthd != EncryptionMethod::Plaintext {
+            let writer = box_try!(EncrypterWriter::new(
+                file.take().unwrap(),
+                mthd,
+                &enc_info.key,
+                box_try!(Iv::from_slice(&enc_info.iv)),
+            ));
+            encrypted_file = Some(writer);
+        }
+    }
+    let mut writer = if let Some(file) = file.as_mut() {
+        file as &mut dyn Write
+    } else {
+        let enc_file = encrypted_file.as_mut().unwrap();
+        enc_file as &mut dyn Write
+    };
+
     let mut stats = BuildStatistics::default();
     box_try!(snap.scan_cf(cf, start_key, end_key, false, |key, value| {
         stats.key_count += 1;
         stats.total_size += key.len() + value.len();
-        box_try!(file.encode_compact_bytes(key));
-        box_try!(file.encode_compact_bytes(value));
+        box_try!(BytesEncoder::encode_compact_bytes(&mut writer, key));
+        box_try!(BytesEncoder::encode_compact_bytes(&mut writer, value));
         Ok(true)
     }));
+
+    let mut file = match file {
```

Comment:
the code would be more readable if we add a `should_encrypt` boolean variable earlier in this function.